sdk/communication/azure-communication-rooms/azure/communication/rooms/_rooms_client.py

```diff
@@ -0,0 +1,300 @@
+    """A client to interact with the AzureCommunicationService Rooms gateway.
+
+    This client provides operations to manage rooms.
+
+    :param str endpoint:
+        The endpoint url for Azure Communication Service resource.
+    :param str credential:
+        The access key we use to authenticate against the service.
+    """
+    def __init__(
+            self, endpoint: str,
+            credential: str,
+            **kwargs
+    ) -> None:
+        try:
+            if not endpoint.lower().startswith('http'):
+                endpoint = "https://" + endpoint
+        except AttributeError:
+            raise ValueError("Account URL must be a string.")
+
+        if not credential:
+            raise ValueError(
+                "invalid credential from connection string.")
+
+        # TokenCredential not supported at the moment
+        if hasattr(credential, "get_token"):
+            raise TypeError("Unsupported credential: {}. Use an access token string to use HMACCredentialsPolicy"
+                    "or a token credential from azure.identity".format(type(credential)))
+
+        self._endpoint = endpoint
+        self._api_version = kwargs.pop("api_version", DEFAULT_VERSION)
+        self._authentication_policy = get_authentication_policy(endpoint, credential)
+        self._rooms_service_client = AzureCommunicationRoomsService(
+            self._endpoint,
+            api_version = self._api_version,
+            authentication_policy=self._authentication_policy,
+            sdk_moniker=SDK_MONIKER,
+            **kwargs)
+
+    @classmethod
+    def from_connection_string(cls, conn_str: str,
+            **kwargs
+    ) -> 'RoomsClient':
+        """Create RoomsClient from a Connection String.
+
+        :param str conn_str:
+            A connection string to an Azure Communication Service resource.
+        :returns: Instance of RoomsClient.
+        :rtype: ~azure.communication.room.RoomsClient
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/Rooms_sample.py
+                :start-after: [START auth_from_connection_string]
+                :end-before: [END auth_from_connection_string]
+                :language: python
+                :dedent: 8
+                :caption: Creating the RoomsClient from a connection string.
+        """
+        endpoint, access_key = parse_connection_str(conn_str)
+
+        return cls(endpoint, access_key, **kwargs)
+
+    @distributed_trace
+    def create_room(
+        self,
+        *,
+        valid_from: Optional[datetime] = None,
+        valid_until: Optional[datetime] = None,
+        room_join_policy: Optional[RoomJoinPolicy] = None,
+        participants: Optional[List[RoomParticipant]]=None,
+        **kwargs
+    ) -> CommunicationRoom:
+        """Create a new room.
+
+        :param valid_from: (Optional)The timestamp from when the room is open for joining. The timestamp is in
+         RFC3339 format: ``yyyy-MM-ddTHH:mm:ssZ``.
```

Comment:
These docstrings look like they've been copied from the service documentation? Please rephrase so that they would be relevant to an SDK developer (e.g. the formatting of an RFC3339 string is not important here).